tools/c7n_azure/c7n_azure/actions.py

```diff
@@ -52,28 +52,60 @@ def update_resource_tags(tag_action, resource, tags):
             if tag_action.manager.type == 'armresource':
                 raise NotImplementedError('Cannot tag generic ARM resources.')
 
+            # build a GenericResource object from the resource string returned by the Azure API
             az_resource = GenericResource.deserialize(resource)
             api_version = tag_action.session.resource_api_version(az_resource.id)
+
+            # set the tags for the VM
```

Comment:
This action isn't VM specific so it could be setting tags for other resource types too. Maybe remove this comment?

Reply:
Same with other VM specific comments added, except for maybe the one about the plan.... although I'm curious how many other resources that affects.